AnnService/inc/Helper/Concurrent.h

```diff
@@ -21,17 +20,25 @@ namespace Concurrent
 class SpinLock
 {
 public:
-    SpinLock();
-    ~SpinLock();
+	SpinLock() = default;
+
+	void Lock() noexcept 
+	{
+		while (m_lock.test_and_set(std::memory_order_acquire))
+		{
+		}
+	}
 
-    void Lock();
-    void Unlock();
+	void Unlock() noexcept 
+	{
+		m_lock.clear(std::memory_order_release);
+	}
 
     SpinLock(const SpinLock&) = delete;
     SpinLock& operator = (const SpinLock&) = delete;
 
 private:
-    std::atomic_flag m_lock;
+    std::atomic_flag m_lock = ATOMIC_FLAG_INIT;
```

Comment:
Personally I don't recommend to use default member initializer since it may become confused and hard to read when there are multiple data members or constructors. But it should be fine here since it is the only data member and all implementation is in same file now.

Reply:
This is the recommended way of in initializing `std::atomic_flag`, that is why i didn't used it in initializer list as it requires copy initialization using the `ATOMIC_FLAG_INIT`